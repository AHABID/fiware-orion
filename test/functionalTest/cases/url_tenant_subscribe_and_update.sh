source harnessFunctions.sh

echo "1. Make a subscription for E/att on tenant t_01"
echo "+++++++++++++++++++++++++++++++++++++++++++++++++++++"
url="/t_01/NGSI10/subscribeContext"
payload='<?xml version="1.0" encoding="UTF-8"?>
<subscribeContextRequest>
  <entityIdList>
    <entityId type="Test" isPattern="false">
      <id>E</id>
    </entityId>
  </entityIdList>
  <attributeList/>
  <reference>http://127.0.0.1:'${LISTENER_PORT}'/notify</reference>
  <duration>PT1H</duration>
  <notifyConditions>
    <notifyCondition>
      <type>ONCHANGE</type>
      <condValueList>
        <condValue>att</condValue>
      </condValueList>
    </notifyCondition>
  </notifyConditions>
</subscribeContextRequest>'
curlXml ${url} "${payload}"
echo
echo


echo "2. Make a subscription for E/att on tenant t_02"
echo "+++++++++++++++++++++++++++++++++++++++++++++++++++++"
url="/t_02/NGSI10/subscribeContext"
payload='<?xml version="1.0" encoding="UTF-8"?>
<subscribeContextRequest>
  <entityIdList>
    <entityId type="Test" isPattern="false">
      <id>E</id>
    </entityId>
  </entityIdList>
  <attributeList/>
  <reference>http://127.0.0.1:'${LISTENER2_PORT}'/notify</reference>
  <duration>PT1H</duration>
  <notifyConditions>
    <notifyCondition>
      <type>ONCHANGE</type>
      <condValueList>
        <condValue>att</condValue>
      </condValueList>
    </notifyCondition>
  </notifyConditions>
</subscribeContextRequest>'
curlXml ${url} "${payload}"
echo
echo


echo "3. Update t_01/E/att"
echo "++++++++++++++++++++"
url="/t_01/NGSI10/updateContext"
payload='<?xml version="1.0" encoding="UTF-8"?>
<updateContextRequest>
  <contextElementList>
    <contextElement>
      <entityId type="Test" isPattern="false">
        <id>E</id>
      </entityId>
      <contextAttributeList>
        <contextAttribute>
          <name>att</name>
          <type>test</type>
          <contextValue>accu_t01</contextValue>
        </contextAttribute>
      </contextAttributeList>
    </contextElement>
  </contextElementList>
  <updateAction>APPEND</updateAction>
</updateContextRequest>'
curlXml ${url} "${payload}"
echo
echo


echo "4. Update t_02/E/att"
echo "++++++++++++++++++++"
url="/t_02/NGSI10/updateContext"
payload='<?xml version="1.0" encoding="UTF-8"?>
<updateContextRequest>
  <contextElementList>
    <contextElement>
      <entityId type="Test" isPattern="false">
        <id>E</id>
      </entityId>
      <contextAttributeList>
        <contextAttribute>
          <name>att</name>
          <type>test</type>
          <contextValue>accu_t02</contextValue>
        </contextAttribute>
      </contextAttributeList>
    </contextElement>
  </contextElementList>
  <updateAction>APPEND</updateAction>
</updateContextRequest>'
curlXml ${url} "${payload}"
echo
echo


echo "5. Dump accumulator t_01"
echo "+++++++++++++++++++++++++"
curl localhost:${LISTENER_PORT}/dump
echo
echo


echo "6. Dump accumulator t_02"
echo "+++++++++++++++++++++++++"
curl localhost:${LISTENER2_PORT}/dump
echo


