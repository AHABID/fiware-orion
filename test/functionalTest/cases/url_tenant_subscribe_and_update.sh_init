source harnessFunctions.sh

dbInit CB
dbTenantInit ${BROKER_DATABASE_NAME}-t_01
dbTenantInit ${BROKER_DATABASE_NAME}-t_02
brokerStart CB 0-255 IPV4 -multiservice url
accumulatorStart localhost ${LISTENER_PORT}
accumulatorStart localhost ${LISTENER2_PORT}

