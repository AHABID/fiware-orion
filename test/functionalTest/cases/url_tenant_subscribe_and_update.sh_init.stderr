+ source harnessFunctions.sh
++ '[' YES '!=' YES ']'
+ dbInit CB
+ role=CB
+ '[' CB == CB ']'
+ echo 'db.dropDatabase()'
+ mongo testharness --quiet
+ dbTenantInit testharness-t_01
+ db=testharness-t_01
+ echo 'db.dropDatabase()'
+ mongo testharness-t_01 --quiet
+ dbTenantInit testharness-t_02
+ db=testharness-t_02
+ echo 'db.dropDatabase()'
+ mongo testharness-t_02 --quiet
+ brokerStart CB 0-255 IPV4 -multiservice url
+ role=CB
+ traceLevels=0-255
+ ipVersion=IPV4
+ shift
+ shift
+ shift
+ extraParams='-multiservice url'
+ '[' CB == '' ']'
+ '[' 0-255 == '' ']'
+ localBrokerStop CB
+ role=CB
+ '[' CB == CB ']'
+ port=9999
++ ps -fe
++ grep 9999
++ wc -l
++ grep contextBroker
+ running_broker=0
+ '[' 0 -ne 0 ']'
+ localBrokerStart CB 0-255 IPV4 -multiservice url
+ role=CB
+ traceLevels=0-255
+ ipVersion=IPV4
+ shift
+ shift
+ shift
+ extraParams='-multiservice url'
+ IPvOption=
+ '[' IPV4 == IPV4 ']'
+ IPvOption=-ipv4
+ '[' CB == CB ']'
+ port=9999
+ CB_START_CMD='contextBroker -vvvvv -harakiri -port 9999 -pidpath /tmp/orion_9999.pid     -db testharness -t 0-255 -ipv4 -multiservice url'
+ '[' '' == '' ']'
+ contextBroker -vvvvv -harakiri -port 9999 -pidpath /tmp/orion_9999.pid -db testharness -t 0-255 -ipv4 -multiservice url
+ sleep 1
++ ps -fe
++ grep 9999
++ wc -l
++ grep contextBroker
+ running_broker=1
+ '[' 1 -ne 1 ']'
+ accumulatorStart localhost 9997
+ bindIp=localhost
+ port=9997
+ '[' -z 9997 ']'
+ accumulatorStop 9997
+ port=9997
+ '[' -z 9997 ']'
++ curl localhost:9997/pid -s -S
curl: (7) couldn't connect to host
+ kill
kill: uso: kill [-s id_señal | -n num_señal | -id_señal] pid | idtrabajo ... ó kill -l [id_señal]
+ sleep 1
++ ps -fe
++ grep 9997
++ wc -l
++ grep accumulator-server
+ running_app=0
+ '[' 0 -ne 0 ']'
+ echo accumulator running as PID 25956
+ port_not_ok=1
+ typeset -i time
+ accumulator-server.py 9997 /notify localhost
+ time=0
+ '[' 1 -eq 0 ']'
+ '[' 0 -eq 30 ']'
+ sleep 1
 * Running on http://localhost:9997/
 * Restarting with reloader
+ time=0+1
+ nc -z localhost 9997
+ port_not_ok=0
+ '[' 0 -eq 0 ']'
+ accumulatorStart localhost 9977
+ bindIp=localhost
+ port=9977
+ '[' -z 9977 ']'
+ accumulatorStop 9977
+ port=9977
+ '[' -z 9977 ']'
++ curl localhost:9977/pid -s -S
curl: (7) couldn't connect to host
+ kill
kill: uso: kill [-s id_señal | -n num_señal | -id_señal] pid | idtrabajo ... ó kill -l [id_señal]
+ sleep 1
++ ps -fe
++ grep 9977
++ wc -l
++ grep accumulator-server
+ running_app=0
+ '[' 0 -ne 0 ']'
+ echo accumulator running as PID 25956
+ port_not_ok=1
+ typeset -i time
+ accumulator-server.py 9977 /notify localhost
+ time=0
+ '[' 1 -eq 0 ']'
+ '[' 0 -eq 30 ']'
+ sleep 1
 * Running on http://localhost:9977/
 * Restarting with reloader
+ time=0+1
+ nc -z localhost 9977
+ port_not_ok=0
+ '[' 0 -eq 0 ']'
127.0.0.1 - - [09/May/2014 16:51:24] "POST /notify HTTP/1.1" 200 -
127.0.0.1 - - [09/May/2014 16:51:24] "POST /notify HTTP/1.1" 200 -
127.0.0.1 - - [09/May/2014 16:51:24] "GET /dump HTTP/1.1" 200 -
127.0.0.1 - - [09/May/2014 16:51:24] "GET /dump HTTP/1.1" 200 -
127.0.0.1 - - [09/May/2014 16:51:24] "GET /pid HTTP/1.1" 200 -
127.0.0.1 - - [09/May/2014 16:51:25] "GET /pid HTTP/1.1" 200 -
